src/main/java/chess/domain/board/ChessBoard.java

```diff
@@ -11,49 +14,168 @@
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 public class ChessBoard {
+    private static final int SINGLE_KING = 1;
+    private static final int DOUBLE_PAWN = 2;
+    private static final Score HALF_PAWN_SCORE = new Score(0.5);
+    private static final int NEW_GAME_COMMAND = 0;
+
+    private final GameInformationDao gameInformationDao;
+    private final ChessGameDao chessGameDao;
     private final Map<Position, Piece> chessBoard = new LinkedHashMap<>();
+    private final GameInformation gameInformation;
 
-    public ChessBoard() {
+    public ChessBoard(ConnectionGenerator connectionGenerator) {
+        this.gameInformationDao = new GameInformationDao(connectionGenerator);
+        this.chessGameDao = new ChessGameDao(connectionGenerator);
         initializeBlackPieces();
         initializeWhitePieces();
+        saveData();
+        this.gameInformation = bringNewGameInformation();
+    }
+
+    public ChessBoard(int gameId, ConnectionGenerator connectionGenerator) {
+        this.gameInformationDao = new GameInformationDao(connectionGenerator);
+        this.chessGameDao = new ChessGameDao(connectionGenerator);
+        bringChessBoard(chessGameDao.findById(gameId));
+        this.gameInformation = bringGameInformation(gameId);
+    }
+
+    public static ChessBoard from(int gameId, ConnectionGenerator connectionGenerator) {
+        if (gameId == NEW_GAME_COMMAND) {
+            return new ChessBoard(connectionGenerator);
+        }
+        return new ChessBoard(gameId, connectionGenerator);
     }
 
     public void move(Position source, Position target) {
-        if (!canMove(source, target)) {
-            throw new IllegalArgumentException("올바르지 않은 이동입니다.");
+        if (!chessBoard.containsKey(source)) {
+            throw new IllegalArgumentException("이동할 수 있는 말이 없습니다.");
         }
         Piece sourcePiece = chessBoard.get(source);
-        chessBoard.put(target, sourcePiece);
-        chessBoard.remove(source);
+        validateTurn(sourcePiece);
+        validateTurn(sourcePiece);
+
+        if (!canMove(sourcePiece, source, target)) {
+            throw new IllegalArgumentException("올바르지 않은 이동입니다.");
+        }
+        updateChessBoard(source, target, sourcePiece);
+    }
+
+    public boolean isKingCaptured() {
+        int kingCount = Math.toIntExact(chessBoard.values().stream()
+                .filter(Piece::isKing)
+                .count());
+        return kingCount == SINGLE_KING;
+    }
+
+    public Color findWinnerByKing() {
+        return chessBoard.values().stream().filter(Piece::isKing)
+                .findFirst()
+                .map(Piece::getColor)
+                .orElseThrow(() -> new UnsupportedOperationException("King이 존재하지 않습니다."));
+    }
+
+    public Score calculateScore(Color color) {
+        double scoreValue = chessBoard.values().stream().filter(piece -> piece.isSameColor(color))
+                .mapToDouble(Piece::getScore).sum();
```

Comment:
stream 사용시 한줄에 하나씩 사용해주세요!
```suggestion
        double scoreValue = chessBoard.values().stream()
            .filter(piece -> piece.isSameColor(color))
            .mapToDouble(Piece::getScore)
            .sum();
```